### PR TITLE
Fix clippy 0.1.68

### PR DESCRIPTION
**Pull request by godofdream**
I fixed plenty of clippy warnings and errors.
I updated the dependencies.
Also I replaced some deprecated functions of the parser, humansize, and chrono.
The only real change is the sorting function, which should be faster now.

### CODE REVIEW

File: src/builtins/filters/array.rs [hunk 1]
```diff
@@ -79,20 +78,34 @@ pub fn sort(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
...
-    })?;
+    // make sure the types are identical, and the field exists
+    let first = arr[0]
+        .pointer(&ptr)
+        .ok_or_else(|| Error::msg(format!("attribute '{attribute}' does not reference a field")))?;
+    let discriminant = std::mem::discriminant(first);
+    let misfits = arr[1..].iter().find_map(|item| 
+        {   
+            let a = item.pointer(&ptr).ok_or(Error::msg(format!("attribute '{attribute}' does not reference a field for {item}")));
+            if a.is_err() {
```
Comment (Keats) [inline src/builtins/filters/array.rs:89]: any reason we don't use `?` above?

**Comment by Keats** [comment]
Let's try to not bump the MSRV though so that means removing the inlined args in format!

**Comment by godofdream** [comment]
what is the current msrv?
I would add it as clippy config

**Comment by Keats** [comment]
https://github.com/Keats/tera/blob/master/.github/workflows/ci.yml#L18

**Comment by godofdream** [comment]
I'm redoing the pullrequest with a cleaned up branch

**Pull request closed without merging**